cmd/agent/app/check.go

```diff
@@ -47,7 +47,7 @@ func init() {
 	checkCmd.Flags().BoolVarP(&checkRate, "check-rate", "r", false, "check rates by running the check twice with a 1sec-pause between the 2 runs")
 	checkCmd.Flags().IntVarP(&checkTimes, "check-times", "t", 1, "number of times to run the check")
 	checkCmd.Flags().IntVar(&checkPause, "pause", 0, "pause between multiple runs of the check, in milliseconds")
-	checkCmd.Flags().StringVarP(&logLevel, "log-level", "l", "", "set the log level (default 'off')")
+	checkCmd.Flags().StringVarP(&logLevel, "log-level", "l", "", "set the log level (default 'off') (deprecated, use the env var DD_LOG_LEVEL instead)")
```

Comment:
We thought it would be too verbose to honor that on short cli commands, so we only check the env var which is a more explicit way of saying "hey, I really wanna see the logs".

Reply:
Makes sense although using config helpers avoid redefining env prefixes and all that stuff. Also I'm particularly thinking of the case of containerized environments and looking back at the change I'm not sure it's really convenient to rely on env var to handle commands logging level. Typically it's easier to invoke an `agent check` and other commands with the right log-level flag than redeploying a container with a new env var value.

Reply:
That's to avoid redefining env prefixes and that kind of stuff but we can live with it. However, I need a clarification on the deprecation notice: do we plan on removing that flag at all? It is particularly useful to keep those flags in containerized environment.

Reply:
The plan is to remove them in a future version of the agent, since you can use the env var instead. That should be as easy to use on containerized environments, plus it works on every command and not only on `jmxfetch` and `check`.

Reply:
It's easier in the sense that if I wanted to run the `agent check <foo>` command and get the debug log but still make sure the agent process outputs info logs for instance I just need to pass the right flag when invoking kubectl exec for instance, instead of changing the whole pod/container environment or doing stuff like `DD_LOG_LEVEL=foo agent <command> ...` but it's a personal preference.

Reply:
I see your point and it certainly makes it a bit more cumbersome to use, but I don't think this is enough reason to add a _third_ way to set the log level. 

We can discuss it elsewhere to not drag this PR out, and if we decide to add this flag (the proper way: app-wide in `app.go` instead of per-command) it can be done in a separate PR.